server/sockets/list.js

```diff
@@ -226,24 +202,33 @@ const updateItem = (
   });
 };
 
-const addComment = socket =>
-  socket.on(CommentActionTypes.ADD_SUCCESS, data => {
-    const { listId } = data;
+const addComment = io => data => {
+  const { listId } = data;
 
-    socket.broadcast
-      .to(listChannel(listId))
-      .emit(CommentActionTypes.ADD_SUCCESS, data);
-  });
+  io.sockets.to(listChannel(listId)).emit(CommentActionTypes.ADD_SUCCESS, data);
+
+  return Promise.resolve();
+};
+
+const cloneItem = (io, listClients, viewersIds) => data => {
+  const { userId, item } = data;
+  const dataToSend = { ...data, ...item };
+
+  viewersIds.forEach(viewerId => {
+    const viewerIdAsString = viewerId.toString();
 
-const cloneItem = socket =>
-  socket.on(ItemActionTypes.CLONE_SUCCESS, data => {
-    const { listId } = data;
+    if (viewerIdAsString !== userId.toString()) {
+      if (listClients.has(viewerIdAsString)) {
+        const { socketId } = listClients.get(viewerIdAsString);
 
-    socket.broadcast
-      .to(listChannel(listId))
-      .emit(ItemActionTypes.CLONE_SUCCESS, data);
+        io.sockets.to(socketId).emit(ItemActionTypes.CLONE_SUCCESS, dataToSend);
```

Comment:
if you check if listClients has specific userId you have to check if he is on this list's view:
```
if (viewerIdAsString !== userId.toString()) {
  if (listClients.has(viewerIdAsString)) {
    const { socketId, viewId } = listClients.get(viewerIdAsString);
    if(viewId === listId) {
        io.sockets.to(socketId).emit(ItemActionTypes.CLONE_SUCCESS, dataToSend);
    }
  }
}
```

or maybe you don't need to check listClients and iterate through viewers but instead of this send data to list room:
```
 io.sockets.to(listChannel(listId).emit(ItemActionTypes.CLONE_SUCCESS, dataToSend);
```
please if you change this change code in other places you refactored

Reply:
1) I can't send it to list room, because the sender will get it TWICE. 
This is why I came with this solution to recognize the sender. The problem was indeed in sending just to list room.

2) Do I really need to check viewId? If the user is not on the list - there would be no userId in listClients map? 
Is there any case, when he can be on other view and still be inside listClients map??

Reply:
he can be on other List, this Map stores all clients on all list's views